rewrite-java/src/main/java/org/openrewrite/java/tree/J.java

```diff
@@ -1067,17 +1080,25 @@ public Case withBody(J body) {
             return getPadding().withBody(JRightPadded.withElement(this.body, body));
         }
 
+        @Nullable
+        @Getter
+        @With
+        Expression guard;
+
         @JsonCreator
-        public Case(UUID id, Space prefix, Markers markers, Type type, @Deprecated @Nullable Expression pattern, JContainer<Expression> expressions, JContainer<Statement> statements, @Nullable JRightPadded<J> body) {
+        public Case(UUID id, Space prefix, Markers markers, Type type, @Deprecated @Nullable Expression pattern, @Nullable JContainer<Expression> expressions, @Nullable JContainer<J> caseLabels, @Nullable Expression guard, JContainer<Statement> statements, @Nullable JRightPadded<J> body) {
             this.id = id;
             this.prefix = prefix;
             this.markers = markers;
             this.type = type;
             if (pattern != null) {
-                this.expressions = requireNonNull(JContainer.withElementsNullable(null, singletonList(pattern)));
+                this.caseLabels = requireNonNull(JContainer.withElementsNullable(null, singletonList(pattern)));
+            } else if (expressions != null && !expressions.getElements().isEmpty()) {
```

Comment:
I think we better leave out the second part so that we know that `caseLabels` will be set to a non-null value.
```suggestion
            } else if (expressions != null) {
```